src/main/scala/com/databricks/labs/overwatch/utils/RefreshNodeDetails.scala

```diff
@@ -0,0 +1,344 @@
+      case _ => throw new Exception(s"Supported Cloud Types are azure|aws|gcp")
+    }
+  }
+
+  // name the local temp files by cloud
+  private def getLocalStaticResourceFileName(cloud: String): String = {
+    cloud.toLowerCase match {
+      case "azure" => "Azure_Instance_Details.csv"
+      case "aws" => "AWS_Instance_Details.csv"
+      case "gcp" => "Gcp_Instance_Details.csv"
+      case _ => throw new Exception(s"Supported Cloud Types are azure|aws|gcp")
+    }
+  }
+
+  // land the raw files from github main branch to the temp dir
+  private def landStaticResourceFiles: this.type = {
+    distinctClouds.foreach(cloudString => {
+      val url = getStaticCloudResourceLink(cloudString)
+      sc.addFile(url)
+      val fileName = getLocalStaticResourceFileName(cloudString)
+      val staticResourcePath = SparkFiles.get(fileName)
+      dbutils.fs.cp(s"file://$staticResourcePath", s"$tempWorkingDir/$fileName")
+    })
+    this
+  }
+
+  // build the dataframe for all clouds in scope from the git hub main branch
+  private def getLatestInstanceDetailsDF: DataFrame = {
+
+    val subsequentClouds = distinctClouds.drop(1)
+    val firstCloud = distinctClouds(0)
+
+    // build df for first cloud in scope
+    val firstCloudInstanceDetailsDF = spark.read.option("header", "true").format("csv")
+      .load(s"$tempWorkingDir/${getLocalStaticResourceFileName(firstCloud)}")
+      .withColumn("organization_id", explode(lit(orgIdsByCloud(firstCloud))))
+
+    // union all subsequent clouds to the first df using foldLeft
+    val rawUpdatesDF = subsequentClouds.foldLeft(firstCloudInstanceDetailsDF)((df, cloudString) => {
+      val filePathX = s"$tempWorkingDir/${getLocalStaticResourceFileName(cloudString)}"
+      val dfX = spark.read.option("header", "true").format("csv")
+        .load(filePathX)
+        .withColumn("organization_id", explode(lit(orgIdsByCloud(cloudString))))
+      df.unionByName(dfX, allowMissingColumns = true)
+    })
+
+    // add additional meta df such as primordial_date and workspace_name
+    rawUpdatesDF
+      .join(originalMetaDF, Seq("organization_id"))
+
+  }
+
+  // build the updates dataframe to be merged into instanceDetails
+  private def deriveMergeDF: DataFrame = {
+
+    val fieldsToUpdate = getLatestInstanceDetailsDF.columns
+      .filterNot(fieldName => keyFields.map(_.toLowerCase).contains(fieldName.toLowerCase))
+
+    // append _update to all fields except the key fields
+    val latestInstanceDetailsFromGitDF = fieldsToUpdate
+      .foldLeft(getLatestInstanceDetailsDF)((df, f) =>
+        df.withColumnRenamed(f, s"${f}_update")
+      )
+
+    // full join to get original df with all values from git where git fields (less keys) are renamed to be
+    // suffixed with _update
+    val mergeUpdatesDF = instanceDetailsDF
+      .join(latestInstanceDetailsFromGitDF, keyFields, "full")
```

Comment:
we should be using left join , instead of full join for a better performance. Also we are not making any changes in the target table, so doing a left join will be more efficient

Reply:
done, thanks.